File: problems/1342-number-of-steps-to-reduce-a-number-to-zero/python/1342_number_of_steps_to_reduce_a_number_to_zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        counter: int = 0
        while num != 0:
            if num % 2 == 0:
                num /= 2
            else:
                num -= 1
            counter += 1
        return counter


def test(num: int) -> None:
    result: int = Solution().numberOfSteps(num)
    print(f"Recutions to zero for {num} is {result}!")


if __name__ == "__main__":
    test(14)
    test(8)
    test(123)

File: problems/1342-number-of-steps-to-reduce-a-number-to-zero/csharp/1342_number_of_steps_to_reduce_a_number_to_zero.cs
using System;

public static class Program
{
  public static void Main(string[] args)
  {
    Solution s = new Solution();
    Console.WriteLine(s.NumberOfSteps(14));
    Console.WriteLine(s.NumberOfSteps(8));
    Console.WriteLine(s.NumberOfSteps(123));
  }
}

public class Solution
{
  public int NumberOfSteps(int number)
  {
    int counter = 0;
    while (number != 0)
    {
      if (number % 2 == 0) number /= 2;
      else number -= 1;
      counter++;
    }
    return counter;
  }
}